src/client/connection.rs

```diff
@@ -165,6 +184,27 @@ impl Connection {
                 self.is_network_enabled = true;
                 Err(true)
             }
+            Err(NetworkError::NetworkStreamClosed) => {
```

Comment:
That's why I tried to filter out the two cases of `is_disconnecting` and gcp. Would you suggest simply returning false and letting the `should_reconnect_again` method figure out whether we should reconnect?

Reply:
Yeah. Even gcloud could have disconnected the client for several reasons, not just jwt expiry (rate limiting is one example). We shouldn't try to judge the nature of disconnection based on connection method. We should just send `Disconnect` over the request channel and chill.

Reply:
Right, I hadn't considered these cases. I'll return `Err(false)` so that `should_reconnect_again` handles the reconnection logic and breaks the reconnection loop in case of `Disconnecting`

Reply:
fixed in 23f60ea